cli/pkg/schema/errors.go

```diff
@@ -0,0 +1,245 @@
+
+	if path.ResourceType == "service" && path.Property == "container" {
+		schemas = append(schemas, "docker", "image")
+	}
+
+	return fmt.Sprintf("Must provide exactly one of: %s", strings.Join(schemas, " OR "))
+}
+
+func (ef *errorFormatter) ShouldSkipError(errType string) bool {
+	return errType == "pattern" && ef.path.ResourceType == "entrypoint" && ef.path.Property == "routes"
+}
+
+func PrettyPrintPattern(pattern *regexp.Regexp) string {
+	patterns := map[string]string{
+		`^(([a-z]+)/([a-z]+)@(\d+)|file:([^\s]+))$`: "Must be in the format: `<team>/<platform>@<revision>` or `file:<path>`",
+	}
+
+	plainTextPattern, ok := patterns[pattern.String()]
+	if !ok {
+		return pattern.String()
+	}
+
+	return plainTextPattern
+}
+
+func (t *NitricErrorTemplate) ErrorFormat() string {
+	return "{{ error_prefix .field}}: {{.description}}"
+}
+
+func (t *NitricErrorTemplate) NumberOneOf() string {
+	return "{{one_of .field}}"
+}
+
+func (t *NitricErrorTemplate) InvalidPropertyName() string {
+	return "{{ invalid_property_name .field}} {{.property}}"
+}
+
+func (t *NitricErrorTemplate) DoesNotMatchPattern() string {
+	return "{{pretty_print_pattern .pattern}}"
+}
+
+func (t *NitricErrorTemplate) Required() string {
+	return "The {{.property}} property is required"
+}
+
+func ErrorTemplateFunc() map[string]interface{} {
+	return map[string]interface{}{
+		"error_prefix": func(field string) string {
+			formatter := newErrorFormatter(field)
+			return formatter.FormatErrorPrefix()
+		},
+		"invalid_property_name": func(field string) string {
+			formatter := newErrorFormatter(field)
+			return formatter.FormatInvalidProperty()
+		},
+		"one_of": func(field string) string {
+			formatter := newErrorFormatter(field)
+			return formatter.FormatNumberOneOf()
+		},
+		"pretty_print_pattern": func(pattern *regexp.Regexp) string {
+			return PrettyPrintPattern(pattern)
+		},
+	}
+}
+
+func GetSchemaValidationErrors(errs []gojsonschema.ResultError) []ValidationError {
+	validationErrors := []ValidationError{}
+
+	for _, err := range errs {
+		formatter := newErrorFormatter(err.Field())
+
+		// Skip certain errors based on context
+		if formatter.ShouldSkipError(err.Type()) {
+			continue
+		}
+
+		yamlContext := GenerateYamlContext(err, err.Description())
+
+		validationErrors = append(validationErrors, ValidationError{
+			Path:        err.Field(),
+			Message:     err.String(),
+			ErrorType:   err.Type(),
+			YamlContext: yamlContext,
+		})
+	}
+
+	slices.SortFunc(validationErrors, func(a, b ValidationError) int {
+		return strings.Compare(a.Path, b.Path)
+	})
+
+	return validationErrors
+}
+
+/*
+	Formats the error messages like this:
+
+---> Error Message
+
+	|
+	| YAML Context
+	|
+*/
+func FormatValidationErrors(errs []ValidationError) string {
+	if len(errs) == 0 {
+		return ""
+	}
+
+	arrow := "--->"
+	linePrefix := "  |"
+	var errsStr strings.Builder
+	for _, err := range errs {
+		errsStr.WriteString(fmt.Sprintf("%s %s\n", arrow, err.Message))
```

Comment:
strings builder also implements `io.Writer` so an alternative to calling Writing string and fmt.Sprinf is:

```suggestion
	fmt.Fprintf(&errsStr, "%s %s\n", arrow, err.Message)
```